src/Database/Esqueleto/Internal/Internal.hs

```diff
@@ -1425,6 +1449,121 @@ data LockingKind
       --
       -- @since 2.2.7
 
+newtype (ComparableLockStrength a) => StrongestLock a =
+  StrongestLock { getStrongestLock :: Maybe a}
+
+class ComparableLockStrength a where
+  -- | Create a lock that is at least as strong as both locks and at most as
+  -- strong as the stronger lock
+  -- Should satisfy:
+  --    If compare a b = GT
+  --        compareLockStrength (unionLocks a b) b = GT
+  --        compareLockStrength a (unionLocks a b) = EQ
+  --    If compare a b = EQ
+  --        compareLockStrength (unionLocks a b) b = EQ
+  --        compareLockStrength a (unionLocks a b) = EQ
+  --    If compare a b = LT
+  --        compareLockStrength (unionLocks a b) b = EQ
+  --        compareLockStrength a (unionLocks a b) = LT
+  unionLocks :: ComparableLockStrength a => a -> a -> a
+  compareLockStrength :: ComparableLockStrength a => a -> a -> Ordering
+
+instance ComparableLockStrength a => Semigroup (StrongestLock a) where
+  (<>) (StrongestLock Nothing) pright = pright
+  (<>) pleft (StrongestLock Nothing) = pleft
+  (<>) (StrongestLock (Just pright)) (StrongestLock (Just pleft)) =
+    case compareLockStrength pleft pright of
+      GT -> StrongestLock (Just pright)
+      EQ -> StrongestLock (Just $ unionLocks pright pleft)
+      LT -> StrongestLock (Just pleft)
```

Comment:
🤔 

```suggestion
  (<>) (StrongestLock (Just pleft)) (StrongestLock (Just pright)) =
    case compareLockStrength pleft pright of
      GT -> StrongestLock (Just pright)
      EQ -> StrongestLock (Just $ unionLocks pright pleft)
      LT -> StrongestLock (Just pleft)
```

Which I think also swaps the other ones.

```
> compare 1 2
LT
```

Reply:
removed this since we can just concatenate the locking clauses for postgres